In [2]:
import pandas as pd

LEYENDO EL ARCHIVO ACCIDENTES AVIONES

In [3]:
aa = pd.read_csv('AccidentesAviones.csv')
aa.shape

(5008, 18)

In [4]:
aa.head(2)

Unnamed: 0               fecha HORA declarada                     Ruta  \
0           0  September 17, 1908           1718      Fort Myer, Virginia   
1           1  September 07, 1909              ?  Juvisy-sur-Orge, France   

               OperadOR flight_no          route           ac_type  \
0  Military - U.S. Army         ?  Demonstration  Wright Flyer III   
1                     ?         ?       Air show    Wright Byplane   

  registration cn_ln all_aboard PASAJEROS A BORDO crew_aboard  \
0            ?     1          2                 1           1   
1          SC1     ?          1                 0           1   

  cantidad de fallecidos passenger_fatalities crew_fatalities ground  \
0                      1                    1               0      0   
1                      1                    0               0      0   

                                             summary  
0  During a demonstration flight, a U.S. Army fly...  
1  Eugene Lefebvre was the first pilot to ever be...

In [5]:
aa.tail(2)

Unnamed: 0          fecha HORA declarada                        Ruta  \
5006        5006  July 04, 2021          11:30  Patikul, Sulu, Philippines   
5007        5007  July 06, 2021           1500              Palana, Russia   

                             OperadOR flight_no                         route  \
5006  Military - Philippine Air Force         ?  Cagayan de Oro-Lumbia - Jolo   
5007    Kamchatka Aviation Enterprise       251        Petropavlovsk - Palana   

                       ac_type registration  cn_ln all_aboard  \
5006  Lockheed C-130H Hercules         5125   5125         96   
5007        Antonov An 26B-100     RA-26085  12310         28   

     PASAJEROS A BORDO crew_aboard cantidad de fallecidos  \
5006                88           8                     50   
5007                22           6                     28   

     passenger_fatalities crew_fatalities ground  \
5006                    ?               ?      3   
5007                   22               6      0   

                                                summary  
5006  While attempting to land at Jolo Airport, the ...  
5007  The passenger plane crashed into the top of a ...

DE LA INFORMACION CONTENIDA EN EL CONJUNTO DE DATOS Y PENSANDO EN EL TABLERO DE CONTROL DESEO OBTENER EL AÑO DEL SINIESTRO LOS PASAJEROS A BORDO Y SI LOS FALLECIDOS HACIAN PARTE DE LA TRIPULACION O ERAN PASAJEROS DE ESE MANERA HAY QUE ELIMINAR LAS COLUMNAS QUE APORTAN OTRO TIPO DE INFORMACION

In [6]:
aa = aa.drop('Unnamed: 0', axis=1)
aa = aa.drop('HORA declarada', axis=1)
aa = aa.drop('summary', axis=1)
aa.head(3)

fecha                       Ruta              OperadOR  \
0  September 17, 1908        Fort Myer, Virginia  Military - U.S. Army   
1  September 07, 1909    Juvisy-sur-Orge, France                     ?   
2       July 12, 1912  Atlantic City, New Jersey  Military - U.S. Navy   

  flight_no          route           ac_type registration cn_ln all_aboard  \
0         ?  Demonstration  Wright Flyer III            ?     1          2   
1         ?       Air show    Wright Byplane          SC1     ?          1   
2         ?    Test flight         Dirigible            ?     ?          5   

  PASAJEROS A BORDO crew_aboard cantidad de fallecidos passenger_fatalities  \
0                 1           1                      1                    1   
1                 0           1                      1                    0   
2                 0           5                      5                    0   

  crew_fatalities ground  
0               0      0  
1               0      0  
2               5      0

ENCONTRANDO REGISTROS DUPLICADOS

In [7]:
registros_duplicados = aa[aa.duplicated()].sum()
registros_duplicados

fecha                     0
Ruta                      0
OperadOR                  0
flight_no                 0
route                     0
ac_type                   0
registration              0
cn_ln                     0
all_aboard                0
PASAJEROS A BORDO         0
crew_aboard               0
cantidad de fallecidos    0
passenger_fatalities      0
crew_fatalities           0
ground                    0
dtype: object

CONJUNTO DE DATOS SIN DUPLICADOS

ME INTERESA CONOCER LA COLUMNA OPERADOR

In [8]:
valores_unicos = aa['OperadOR'].unique()
valores_unicos

array(['Military - U.S. Army', '?', 'Military - U.S. Navy', ...,
       'South Sudan Supreme Airlines', 'Soloy Helicopters',
       'Kamchatka Aviation Enterprise'], dtype=object)

In [9]:
conteo_operadores = aa['OperadOR'].value_counts()
conteo_operadores

OperadOR
Aeroflot                            253
Military - U.S. Air Force           141
Air France                           74
Deutsche Lufthansa                   63
United Air Lines                     44
                                   ... 
Sierra Pacific Airlines               1
Pathet Lao Airlines                   1
Military - West German Air Force      1
Cessnyca                              1
Kamchatka Aviation Enterprise         1
Name: count, Length: 2268, dtype: int64

In [10]:
conteo_operadores[conteo_operadores == 1].sum()

1734

 DEL DATAFRAME AQUELLOS OPERADORES QUE APARECEN UNA SOLA VEZ, EN EL TABLERO DE CONTROL NUNCA SE MOSTRARAN, LOS AGRUPO BAJO EL TITULO DE OTROS

In [11]:
operadores_a_renombrar = conteo_operadores[conteo_operadores == 1].index

def renombrar_operadores(operador):
    if operador in operadores_a_renombrar:
        return 'Others'
    else:
        return operador

aa['OperadOR'] = aa['OperadOR'].apply(renombrar_operadores)

aa.tail(2)


fecha                        Ruta  \
5006  July 04, 2021  Patikul, Sulu, Philippines   
5007  July 06, 2021              Palana, Russia   

                             OperadOR flight_no                         route  \
5006  Military - Philippine Air Force         ?  Cagayan de Oro-Lumbia - Jolo   
5007                           Others       251        Petropavlovsk - Palana   

                       ac_type registration  cn_ln all_aboard  \
5006  Lockheed C-130H Hercules         5125   5125         96   
5007        Antonov An 26B-100     RA-26085  12310         28   

     PASAJEROS A BORDO crew_aboard cantidad de fallecidos  \
5006                88           8                     50   
5007                22           6                     28   

     passenger_fatalities crew_fatalities ground  
5006                    ?               ?      3  
5007                   22               6      0

In [12]:
conteo_operadores = aa['OperadOR'].value_counts()
conteo_operadores

OperadOR
Others                                  1734
Aeroflot                                 253
Military - U.S. Air Force                141
Air France                                74
Deutsche Lufthansa                        63
                                        ... 
SATCO                                      2
Aden Airways                               2
Frontier Airlines                          2
British Eagle International Airlines       2
Military - Myanmar Air Force               2
Name: count, Length: 535, dtype: int64

In [13]:
def conservar_sindato(texto):
    if '?' in texto:
        return 'Sin_Dato'
    else:
        return texto

aa['OperadOR'] = aa['OperadOR'].apply(conservar_sindato)

DE LA COLUMNA OPERADOR AHORA ES MAS VISIBLE AQUELLOS CON MAS ACCIDENTES

CONOCIENDO LA COLUMNA CN_IN, REGISTRATION Y AC_TYPE

In [14]:
conteo_cn_In = aa['cn_ln'].value_counts()
conteo_cn_In

cn_ln
?        667
1          8
4          8
125        7
229        6
        ... 
068        1
1142       1
43559      1
10182      1
12310      1
Name: count, Length: 3908, dtype: int64

In [15]:
conteo_registration = aa['registration'].value_counts()
conteo_registration

registration
?             272
49              3
SU-AFK          2
2               2
19              2
             ... 
CCCP-61663      1
EC-ANR          1
N1301N          1
CS-THB          1
RA-26085        1
Name: count, Length: 4701, dtype: int64

LAS COLUMNAS A ELIMINAR SON REFERENTES A LA MATRICULA DE LA AERONAVE, NO APORTAN INFORMACION PARA EL TABLERO DE CONTROL

In [16]:
aa = aa.drop('cn_ln', axis=1)
aa = aa.drop('registration', axis=1)
aa.head(1)

fecha                 Ruta              OperadOR flight_no  \
0  September 17, 1908  Fort Myer, Virginia  Military - U.S. Army         ?   

           route           ac_type all_aboard PASAJEROS A BORDO crew_aboard  \
0  Demonstration  Wright Flyer III          2                 1           1   

  cantidad de fallecidos passenger_fatalities crew_fatalities ground  
0                      1                    1               0      0

LA DECISION QUE TOMO ES DE LA COLUMNA AC_TYPE ESTANDARIZAR EL FABRICANTE PARA PODER GRAFICAR LA COLUMNA DE MEJOR MANERA. HAY DEMASIADAS REFERENCIAS DE AERONAVES Y NO ES OPTIMO, AL FINAL AQUELLOS FABRICANTES CON MENOS DE 10 APARICIONES SE AGRUPARAN EN OTROS

In [17]:
conteo_ac_type = aa['ac_type'].value_counts()
conteo_ac_type

ac_type
Douglas DC-3                                333
de Havilland Canada DHC-6 Twin Otter 300     81
Douglas C-47A                                70
Douglas C-47                                 64
Douglas DC-4                                 41
                                           ... 
Vickers Viscount 759D                         1
H-21B                                         1
Avro 685 York C1                              1
Piper PA-24-250 Comanche                      1
Antonov An 26B-100                            1
Name: count, Length: 2469, dtype: int64

In [18]:
conteo_ac_type[conteo_ac_type > 25]

ac_type
Douglas DC-3                                333
de Havilland Canada DHC-6 Twin Otter 300     81
Douglas C-47A                                70
Douglas C-47                                 64
Douglas DC-4                                 41
Antonov AN-26                                35
Yakovlev YAK-40                              35
Junkers JU-52/3m                             30
De Havilland DH-4                            27
Douglas C-47B                                27
Douglas DC-6B                                27
Name: count, dtype: int64

In [19]:
def conservar_havilland(texto):
    if 'Havilland' in texto:
        return 'Havilland'
    else:
        return texto

aa['ac_type'] = aa['ac_type'].apply(conservar_havilland)

In [20]:
def conservar_convair(texto):
    if 'Convair' in texto:
        return 'Convair'
    else:
        return texto

aa['ac_type'] = aa['ac_type'].apply(conservar_convair)

In [21]:
def conservar_yakovlev(texto):
    if 'Yakovlev' in texto:
        return 'Yakovlev'
    else:
        return texto

aa['ac_type'] = aa['ac_type'].apply(conservar_yakovlev)

In [22]:
def conservar_v(texto):
    if 'Vickers' in texto:
        return 'Vickers'
    else:
        return texto

aa['ac_type'] = aa['ac_type'].apply(conservar_v)

In [23]:
def conservar_douglas(texto):
    if 'Douglas' in texto:
        return 'Douglas'
    else:
        return texto

aa['ac_type'] = aa['ac_type'].apply(conservar_douglas)

In [24]:
def conservar_a(texto):
    if 'Antonov' in texto:
        return 'Antonov'
    else:
        return texto

aa['ac_type'] = aa['ac_type'].apply(conservar_a)

In [25]:
def conservar_b(texto):
    if 'Boeing' in texto:
        return 'Boeing'
    else:
        return texto

aa['ac_type'] = aa['ac_type'].apply(conservar_b)

In [26]:
def conservar_l(texto):
    if 'Lockheed' in texto:
        return 'Lockheed'
    else:
        return texto

aa['ac_type'] = aa['ac_type'].apply(conservar_l)

In [27]:
def conservar_c(texto):
    if 'Curtis' in texto:
        return 'Curtiss'
    else:
        return texto

aa['ac_type'] = aa['ac_type'].apply(conservar_c)

In [28]:
def conservar_i(texto):
    if 'Ilyushin' in texto:
        return 'Ilyushin'
    else:
        return texto

aa['ac_type'] = aa['ac_type'].apply(conservar_i)

In [29]:
def conservar_f(texto):
    if 'Fokker' in texto:
        return 'Fokker'
    else:
        return texto

aa['ac_type'] = aa['ac_type'].apply(conservar_f)

In [30]:
def conservar_ju(texto):
    if 'Junkers' in texto:
        return 'Junkers'
    else:
        return texto

aa['ac_type'] = aa['ac_type'].apply(conservar_ju)

In [31]:
conteo_ac_type = aa['ac_type'].value_counts()
conteo_ac_type

ac_type
Douglas                          1125
Boeing                            411
Lockheed                          326
Havilland                         290
Antonov                           286
                                 ... 
Beechcraft H50                      1
Chance Vought F-8E                  1
Bristol Britannia 102               1
Sud Aviation SE-210 Caravelle       1
Beechcraft B300 King Air 350i       1
Name: count, Length: 1261, dtype: int64

In [32]:
conteo_ac_type[conteo_ac_type < 17]

ac_type
Embraer 110P1 Bandeirante        16
Tupolev TU-134A                  16
CASA 212 Aviocar 200             13
?                                13
Let 410UVP                       12
                                 ..
Beechcraft H50                    1
Chance Vought F-8E                1
Bristol Britannia 102             1
Sud Aviation SE-210 Caravelle     1
Beechcraft B300 King Air 350i     1
Name: count, Length: 1248, dtype: int64

In [34]:
fabricantes_a_renombrar = conteo_ac_type[conteo_ac_type == 1].index

def renombrar_fabricantes(fabricante):
    if fabricante in fabricantes_a_renombrar:
        return 'Others'
    else:
        return fabricante

aa['ac_type'] = aa['ac_type'].apply(renombrar_fabricantes)

aa.tail(3)

fecha                        Ruta  \
5005  June 10, 2021  Near Pyin Oo Lwin, Myanmar   
5006  July 04, 2021  Patikul, Sulu, Philippines   
5007  July 06, 2021              Palana, Russia   

                             OperadOR flight_no                         route  \
5005     Military - Myanmar Air Force         ?          Naypyidaw - Anisakan   
5006  Military - Philippine Air Force         ?  Cagayan de Oro-Lumbia - Jolo   
5007                           Others       251        Petropavlovsk - Palana   

               ac_type all_aboard PASAJEROS A BORDO crew_aboard  \
5005  Beechcraft 1900D         14                12           2   
5006          Lockheed         96                88           8   
5007           Antonov         28                22           6   

     cantidad de fallecidos passenger_fatalities crew_fatalities ground  
5005                     12                   11               1      0  
5006                     50                    ?               ?      3  
5007                     28                   22               6      0

In [35]:
def conservar_sindato(texto):
    if '?' in texto:
        return 'Sin_Dato'
    else:
        return texto

aa['ac_type'] = aa['ac_type'].apply(conservar_sindato)

In [36]:
conteo_ac_type = aa['ac_type'].value_counts()
conteo_ac_type

ac_type
Douglas                             1125
Others                              1005
Boeing                               411
Lockheed                             326
Havilland                            290
                                    ... 
Piper PA-31                            2
BAC One-Eleven 515FB                   2
Beech E18S                             2
NAMC YS-11A-211                        2
Britten Norman BN-2A-27 Islander       2
Name: count, Length: 257, dtype: int64

In [37]:
aa.head(2)

fecha                     Ruta              OperadOR  \
0  September 17, 1908      Fort Myer, Virginia  Military - U.S. Army   
1  September 07, 1909  Juvisy-sur-Orge, France              Sin_Dato   

  flight_no          route ac_type all_aboard PASAJEROS A BORDO crew_aboard  \
0         ?  Demonstration  Others          2                 1           1   
1         ?       Air show  Others          1                 0           1   

  cantidad de fallecidos passenger_fatalities crew_fatalities ground  
0                      1                    1               0      0  
1                      1                    0               0      0

CONOCIENDO LA COLUMNA RUTA AGRUPARE POR LOS PAISES CON MAS SINIESTROS AEREOS DE LA MISMA MANERA QUE CON OTRAS COLUMNAS AQUELLOS PAISES CON BAJA FRECUENCIA SE AGRUPARAN BAJO EL TITULO DE OTROS POR TENER POCA RELEVANCIA EN EL TABLERO DE CONTROL

In [38]:
conteo_ruta = aa['Ruta'].value_counts()
conteo_ruta

Ruta
Moscow, Russia                                         16
Manila, Philippines                                    15
New York, New York                                     14
Sao Paulo, Brazil                                      13
Cairo, Egypt                                           13
                                                       ..
NAS Argentia, Placentia Bay,  Newfoundland,  Canada     1
Anzio, Italy                                            1
Nipe Bay, Cuba                                          1
Near Roccatamburo di Poggiodomo, Italy                  1
Palana, Russia                                          1
Name: count, Length: 4125, dtype: int64

In [125]:
def conservar_ru(texto):
    if 'USSR' in texto:
        return 'Russia'
    else:
        return texto

aa['Ruta'] = aa['Ruta'].apply(conservar_ru)

In [40]:
def conservar_ph(texto):
    if 'Philippines' in texto:
        return 'Philippines'
    else:
        return texto

aa['Ruta'] = aa['Ruta'].apply(conservar_ph)

In [41]:
def conservar_br(texto):
    if 'Brazil' in texto:
        return 'Brazil'
    else:
        return texto

aa['Ruta'] = aa['Ruta'].apply(conservar_br)

In [42]:
def conservar_eg(texto):
    if 'Egypt' in texto:
        return 'Egypt'
    else:
        return texto

aa['Ruta'] = aa['Ruta'].apply(conservar_eg)

In [43]:
def conservar_ge(texto):
    if 'Germany' in texto:
        return 'Germany'
    else:
        return texto

aa['Ruta'] = aa['Ruta'].apply(conservar_ge)

In [44]:
def conservar_ma(texto):
    if 'Maryland' in texto:
        return 'EEUU'
    else:
        return texto

aa['Ruta'] = aa['Ruta'].apply(conservar_ma)

In [45]:
def conservar_vi(texto):
    if 'Virginia' in texto:
        return 'EEUU'
    else:
        return texto

aa['Ruta'] = aa['Ruta'].apply(conservar_vi)

In [46]:
def conservar_mas(texto):
    if 'Massachusetts' in texto:
        return 'EEUU'
    else:
        return texto

aa['Ruta'] = aa['Ruta'].apply(conservar_mas)

In [47]:
def conservar_new(texto):
    if 'New York' in texto:
        return 'EEUU'
    else:
        return texto

aa['Ruta'] = aa['Ruta'].apply(conservar_new)

In [48]:
def conservar_eng(texto):
    if 'England' in texto:
        return 'England'
    else:
        return texto

aa['Ruta'] = aa['Ruta'].apply(conservar_eng)

In [191]:
def conservar_italy(texto):
    if 'italy' in texto:
        return 'Italy'
    else:
        return texto

aa['Ruta'] = aa['Ruta'].apply(conservar_italy)

In [186]:
def conservar_canada(texto):
    if 'Newfoundland' in texto:
        return 'Canada'
    else:
        return texto

aa['Ruta'] = aa['Ruta'].apply(conservar_canada)

In [188]:
conteo_ruta = aa['Ruta'].value_counts()
conteo_ruta.head(10)

Ruta
EEUU         919
Russia       325
Brazil       176
Colombia     153
Canada       146
France       128
India        113
Mexico       105
England      101
Indonesia     99
Name: count, dtype: int64

In [185]:
conteo_ruta[conteo_ruta == 2]

Ruta
Near Lisbon, Portugal        2
Near Rijeka, Yugoslavia      2
Near Gander, Newfoundland    2
Near Brazzaville, Congo      2
Atlanta, Georgia             2
                            ..
Bishkek, Kyrgyzstan          2
Benghazi, Libya              2
Near Lima, Peru              2
South Atlantic Ocean         2
Almeria, Spain               2
Name: count, Length: 113, dtype: int64

In [189]:
ruta_a_renombrar = conteo_ruta[conteo_ruta < 60].index

def renombrar_ruta(ruta):
    if ruta in ruta_a_renombrar:
        return 'Others'
    else:
        return ruta

aa['Ruta'] = aa['Ruta'].apply(renombrar_ruta)

aa.tail(3)

fecha         Ruta                         OperadOR flight_no  \
5005  June 10, 2021       Others     Military - Myanmar Air Force         ?   
5006  July 04, 2021  Philippines  Military - Philippine Air Force         ?   
5007  July 06, 2021       Russia                           Others       251   

                             route           ac_type all_aboard  \
5005          Naypyidaw - Anisakan  Beechcraft 1900D         14   
5006  Cagayan de Oro-Lumbia - Jolo          Lockheed         96   
5007        Petropavlovsk - Palana           Antonov         28   

     PASAJEROS A BORDO crew_aboard cantidad de fallecidos  \
5005                12           2                     12   
5006                88           8                     50   
5007                22           6                     28   

     passenger_fatalities crew_fatalities ground  
5005                   11               1      0  
5006                    ?               ?      3  
5007                   22               6      0

In [192]:
conteo_ruta = aa['Ruta'].value_counts()
conteo_ruta

Ruta
Others         2397
EEUU            919
Russia          325
Brazil          176
Colombia        153
Canada          146
France          128
India           113
Mexico          105
England         101
Indonesia        99
China            97
Germany          90
Italy            86
Philippines      73
Name: count, dtype: int64

CONOCIENDO COLUMNA ROUTE

In [193]:
conteo_route = aa['route'].value_counts()
conteo_route

route
?                                       762
Training                                 96
Sightseeing                              31
Test flight                              23
Sao Paulo - Rio de Janeiro                7
                                       ... 
Prague - Zurich                           1
Temuco - Santiago                         1
Hong Kong - Tainan AFB                    1
Fort Lamy, Chad - Marseilles, France      1
Petropavlovsk - Palana                    1
Name: count, Length: 3838, dtype: int64

In [194]:
aa['route'] = aa['route'].astype(str)

In [195]:
def conservar_sindato(texto):
    if '?' in texto:
        return 'Sin_Dato'
    else:
        return texto

aa['route'] = aa['route'].apply(conservar_sindato)

In [196]:
conteo_route = aa['route'].value_counts()
conteo_route

route
Sin_Dato                      768
Training                       96
Sightseeing                    31
Test flight                    23
Sao Paulo - Rio de Janeiro      7
                             ... 
San Antonio-Maiquetia           1
Rio de Janerio - Belem          1
Prague - Zurich                 1
Temuco - Santiago               1
Petropavlovsk - Palana          1
Name: count, Length: 3833, dtype: int64

In [197]:
conteo_route[conteo_route == 1]

route
Bloemfontein - Johannesburg        1
Kutaisi - Poti                     1
Vitória da Conquista - Salvador    1
Parachuting                        1
Beni - La Paz                      1
                                  ..
San Antonio-Maiquetia              1
Rio de Janerio - Belem             1
Prague - Zurich                    1
Temuco - Santiago                  1
Petropavlovsk - Palana             1
Name: count, Length: 3625, dtype: int64

LA COLUMNA ROUTE TIENE 3625 VALORES UNICOS DE MANERA QUE NO SE PUEDE UTILIZAR PARA ANALISIS DE DATOS Y TOMAR UNA DECISION AL RESPECTO PUES NO ENTREGA PATRON ALGUNO. NO SE ELIMINA PARA NO PERDER DATOS INTERESANTES QUE SON TRAINING SIGHTSEEING Y TEST FLIGHT                     

In [198]:
aa.head(1)

fecha  Ruta              OperadOR flight_no          route  \
0  September 17, 1908  EEUU  Military - U.S. Army         ?  Demonstration   

  ac_type all_aboard PASAJEROS A BORDO crew_aboard cantidad de fallecidos  \
0  Others          2                 1           1                      1   

  passenger_fatalities crew_fatalities ground  
0                    1               0      0

CONOCIENDO LA COLUMNA FLIGHT_NO

In [199]:
conteo_flight = aa['flight_no'].value_counts()
conteo_flight

flight_no
?        3682
-          36
1          11
101         9
6           7
         ... 
138         1
124         1
1-10        1
4422        1
SJ182       1
Name: count, Length: 893, dtype: int64

SE PROCEDE A ELIMINAR LA COLUMNA FLIGHT_NO LOS DATOS NO APORTAN INFORMACION CLAVE

In [200]:
aa = aa.drop('flight_no', axis=1)

In [203]:
aa.head(2)

fecha    Ruta              OperadOR          route ac_type  \
0  September 17, 1908    EEUU  Military - U.S. Army  Demonstration  Others   
1  September 07, 1909  France              Sin_Dato       Air show  Others   

  all_aboard PASAJEROS A BORDO crew_aboard cantidad de fallecidos  \
0          2                 1           1                      1   
1          1                 0           1                      1   

  passenger_fatalities crew_fatalities ground  
0                    1               0      0  
1                    0               0      0

COLUMNA FECHA EXTRAYENDO EL AÑO EN LA COLUMNA YEAR

In [204]:
aa['fecha'] = pd.to_datetime(aa['fecha'])
aa['year'] = aa['fecha'].dt.year
aa = aa.drop('fecha', axis = 1)
aa.head(1)

Ruta              OperadOR          route ac_type all_aboard  \
0  EEUU  Military - U.S. Army  Demonstration  Others          2   

  PASAJEROS A BORDO crew_aboard cantidad de fallecidos passenger_fatalities  \
0                 1           1                      1                    1   

  crew_fatalities ground  year  
0               0      0  1908

In [205]:
conteo_year = aa['year'].value_counts()
conteo_year

year
1946    88
1989    83
1947    82
1948    78
1962    78
        ..
1913     3
1915     2
1909     1
1912     1
1908     1
Name: count, Length: 111, dtype: int64

In [206]:
aa = aa.sort_values(by='year', ascending=True)
aa.head(3)

Ruta              OperadOR          route ac_type all_aboard  \
0    EEUU  Military - U.S. Army  Demonstration  Others          2   
1  France              Sin_Dato       Air show  Others          1   
2    EEUU  Military - U.S. Navy    Test flight  Others          5   

  PASAJEROS A BORDO crew_aboard cantidad de fallecidos passenger_fatalities  \
0                 1           1                      1                    1   
1                 0           1                      1                    0   
2                 0           5                      5                    0   

  crew_fatalities ground  year  
0               0      0  1908  
1               0      0  1909  
2               5      0  1912

In [207]:
def asignar_decada(year):
    if 1908 <= year <= 1917:
        return 1
    elif 1918 <= year <= 1927:
        return 2
    elif 1928 <= year <= 1937:
        return 3
    elif 1938 <= year <= 1947:
        return 4
    elif 1948 <= year <= 1957:
        return 5
    elif 1958 <= year <= 1967:
        return 6
    elif 1968 <= year <= 1977:
        return 7
    elif 1978 <= year <= 1987:
        return 8
    elif 1988 <= year <= 1997:
        return 9
    elif 1998 <= year <= 2007:
        return 10
    elif 2008 <= year <= 2017:
        return 11
    elif 2018 <= year <= 2027:
        return 12
    else:
        return None  # Si el año no cae en ninguna década definida, se asigna None

# Aplicar la función a la columna 'year' para crear la columna 'decade'
aa['decade'] = aa['year'].apply(asignar_decada)

# Imprimir el DataFrame resultante
aa.head(3)


Ruta              OperadOR          route ac_type all_aboard  \
0    EEUU  Military - U.S. Army  Demonstration  Others          2   
1  France              Sin_Dato       Air show  Others          1   
2    EEUU  Military - U.S. Navy    Test flight  Others          5   

  PASAJEROS A BORDO crew_aboard cantidad de fallecidos passenger_fatalities  \
0                 1           1                      1                    1   
1                 0           1                      1                    0   
2                 0           5                      5                    0   

  crew_fatalities ground  year  decade  
0               0      0  1908       1  
1               0      0  1909       1  
2               5      0  1912       1

LA COLUMNA CANTIDAD DE FALLECIDOS

In [254]:
conteofallecidos = aa['cantidad de fallecidos'].value_counts()
conteofallecidos

cantidad de fallecidos
1      384
2      377
3      363
4      242
5      235
      ... 
144      1
261      1
223      1
158      1
150      1
Name: count, Length: 200, dtype: int64

In [275]:
registros_con_interrogante = aa[aa['cantidad de fallecidos'] == '?']
registros_con_interrogante

Ruta                             OperadOR                route  \
99    France                                CIDNA             Sin_Dato   
331    China  China National Aviation Corporation             Sin_Dato   
362    China  China National Aviation Corporation             Sin_Dato   
424    China  China National Aviation Corporation             Sin_Dato   
632    China  China National Aviation Corporation             Sin_Dato   
724   Others       Military - U.S. Army Air Corps             Sin_Dato   
889    China  China National Aviation Corporation  Chunking - Shanghai   
1483  Others                           Aigle Azur             Sin_Dato   

             ac_type all_aboard PASAJEROS A BORDO crew_aboard  \
99            Others          ?                 ?           ?   
331   Sikorsky S-38B          ?                 ?           ?   
362   Sikorsky S-38B          ?                 ?           ?   
424          Douglas          ?                 ?           ?   
632          Douglas          ?                 ?           ?   
724         Sin_Dato          ?                 ?           ?   
889         Sin_Dato          ?                 ?           ?   
1483          Others          ?                 ?           ?   

     cantidad de fallecidos passenger_fatalities crew_fatalities ground  year  \
99                        ?                    ?               ?      ?  1925   
331                       ?                    ?               ?      ?  1934   
362                       ?                    ?               ?      ?  1935   
424                       ?                    ?               ?      ?  1936   
632                       ?                    ?               ?      ?  1942   
724                       ?                    ?               ?      ?  1944   
889                       ?                    ?               ?      ?  1946   
1483                      ?                    ?               ?      ?  1954   

      decade  
99         2  
331        3  
362        3  
424        3  
632        4  
724        4  
889        4  
1483       5

SE PROCEDE A ELIMINAR TODOS LOS REGISTROS DE LA COLUMNA CANTIDAD DE FALLECIDOS CON INTERROGANTE PUES NO SE PUEDEN IMPUTAR DE NINGUNA MANERA PORQUE SUS COLUMNAS NUMERICAS A LA IZQUIERDA TAMBIEN TIENEN INTERROGANTE

In [277]:
aa.drop(aa[aa['cantidad de fallecidos'] == '?'].index, inplace=True)

In [278]:
registros_con_interrogante = aa[aa['cantidad de fallecidos'] == '?']
registros_con_interrogante

Empty DataFrame
Columns: [Ruta, OperadOR, route, ac_type, all_aboard, PASAJEROS A BORDO, crew_aboard, cantidad de fallecidos, passenger_fatalities, crew_fatalities, ground, year, decade]
Index: []

In [279]:
registros_con_interrogante = aa[aa['all_aboard'] == '?']
registros_con_interrogante

Ruta               OperadOR                            route  \
27      EEUU                 Others                          Shuttle   
1275  Others                 Others                         Sin_Dato   
1326  Others                 Others                         Sin_Dato   
1432  Brazil                   REAL                         Sin_Dato   
1597    EEUU                 Others                         Sin_Dato   
1641  Others                 Others                         Sin_Dato   
1669  Others             Air France  Tehran, Iran - Istanbul, Turkey   
3599   China                 Others                         Sin_Dato   
3786    EEUU  Aerolineas Argentinas                         Sin_Dato   

       ac_type all_aboard PASAJEROS A BORDO crew_aboard  \
27      Others          ?                 ?           ?   
1275   Douglas          ?                 ?           ?   
1326   Douglas          ?                 ?           ?   
1432    Others          ?                 ?           ?   
1597   Douglas          ?                 ?           ?   
1641    Others          ?                 ?           ?   
1669  Lockheed          ?                 ?           ?   
3599  Bell 412          ?                 ?           ?   
3786    Boeing          ?                 ?           ?   

     cantidad de fallecidos passenger_fatalities crew_fatalities ground  year  \
27                        3                    2               1     10  1919   
1275                     16                    ?               ?      ?  1951   
1326                      9                    ?               ?      ?  1952   
1432                     26                    ?               ?      ?  1953   
1597                      1                    1               ?      0  1956   
1641                      6                    ?               ?      ?  1957   
1669                      1                    1               0      0  1957   
3599                     10                    ?               ?      0  1989   
3786                      1                    ?               ?      0  1992   

      decade  
27         2  
1275       5  
1326       5  
1432       5  
1597       5  
1641       5  
1669       5  
3599       9  
3786       9

SE ELIMINAN LOS INDICES DONDE PASSENGER_FATALITIES Y CREW_FATALITIES SON INTERROGANTES, NO SE IMPUTAN PORQUE MODIFICA LOS KPI'S TOMO LA DECISION DE NO ASUMIR DATOS COMO POR EJEMPLO EN EL INDICE 1275 DONDE MURIERON 16 PERSONAS PERO DESCONOZCO SI ERAN TRIPULANTES O PASAJEROS

In [280]:
indices_a_eliminar = [1275, 1326, 1432, 1641, 3599, 3786]
aa = aa.drop(indices_a_eliminar)

In [281]:
registros_con_interrogante = aa[aa['all_aboard'] == '?']
registros_con_interrogante

Ruta    OperadOR                            route   ac_type  \
27      EEUU      Others                          Shuttle    Others   
1597    EEUU      Others                         Sin_Dato   Douglas   
1669  Others  Air France  Tehran, Iran - Istanbul, Turkey  Lockheed   

     all_aboard PASAJEROS A BORDO crew_aboard cantidad de fallecidos  \
27            ?                 ?           ?                      3   
1597          ?                 ?           ?                      1   
1669          ?                 ?           ?                      1   

     passenger_fatalities crew_fatalities ground  year  decade  
27                      2               1     10  1919       2  
1597                    1               ?      0  1956       5  
1669                    1               0      0  1957       5

In [282]:
aa.at[27, 'all_aboard'] = 'Sin_Dato'
aa.at[27, 'PASAJEROS A BORDO'] = 'Sin_Dato'
aa.at[27, 'crew_aboard'] = 'Sin_Dato'

In [284]:
aa.at[1597, 'all_aboard'] = 'Sin_Dato'
aa.at[1597, 'PASAJEROS A BORDO'] = 'Sin_Dato'
aa.at[1597, 'crew_aboard'] = 'Sin_Dato'
aa.at[1597, 'crew_fatalities'] = 0

In [285]:
aa.at[1669, 'all_aboard'] = 'Sin_Dato'
aa.at[1669, 'PASAJEROS A BORDO'] = 'Sin_Dato'
aa.at[1669, 'crew_aboard'] = 'Sin_Dato'

In [286]:
registros_con_interrogante = aa[aa['all_aboard'] == '?']
registros_con_interrogante

Empty DataFrame
Columns: [Ruta, OperadOR, route, ac_type, all_aboard, PASAJEROS A BORDO, crew_aboard, cantidad de fallecidos, passenger_fatalities, crew_fatalities, ground, year, decade]
Index: []

In [291]:
registros_con_interrogante = aa[aa['crew_fatalities'] == '?']
registros_con_interrogante

Ruta                         OperadOR  \
4          Others           Military - German Navy   
5         Germany           Military - German Navy   
7         Germany           Military - German Navy   
8         England           Military - German Navy   
9         Germany           Military - German Army   
...           ...                              ...   
4833       Others    Military - Sudanese Air Force   
4847       Others                           Others   
4892       Others                           Others   
4940       Others                           Others   
5001  Philippines  Military - Philippine Air Force   

                             route   ac_type all_aboard PASAJEROS A BORDO  \
4                         Sin_Dato    Others         20                 ?   
5                         Sin_Dato    Others         28                 ?   
7                         Sin_Dato    Others         19                 ?   
8                         Sin_Dato    Others         20                 ?   
9                         Sin_Dato    Others         28                 ?   
...                            ...       ...        ...               ...   
4833                      Sin_Dato   Antonov         22                16   
4847                      Sin_Dato    Others         12                 ?   
4892                      Training    Others         19                21   
4940          Mogadishu - Djibouti    Others         72                72   
5001  Cagayan de Oro-Lumbia - Jolo  Lockheed         96                88   

     crew_aboard cantidad de fallecidos passenger_fatalities crew_fatalities  \
4              ?                     14                    ?               ?   
5              ?                     28                    ?               ?   
7              ?                     19                    ?               ?   
8              ?                     20                    ?               ?   
9              ?                     27                    ?               ?   
...          ...                    ...                  ...             ...   
4833           6                     15                    ?               ?   
4847           ?                     12                    ?               ?   
4892           2                     18                    ?               ?   
4940           ?                      1                    1               ?   
5001           8                     50                    ?               ?   

     ground  year  decade  
4         0  1913       1  
5         0  1913       1  
7         0  1915       1  
8         0  1916       1  
9         0  1916       1  
...     ...   ...     ...  
4833      0  2012      11  
4847      4  2012      11  
4892      0  2014      11  
4940      0  2016      11  
5001      3  2021      12  

[220 rows x 13 columns]

In [292]:
registros_con_interrogante = aa[aa['passenger_fatalities'] == '?']
registros_con_interrogante

Ruta                         OperadOR  \
4          Others           Military - German Navy   
5         Germany           Military - German Navy   
7         Germany           Military - German Navy   
8         England           Military - German Navy   
9         Germany           Military - German Army   
...           ...                              ...   
4830        India                           Others   
4833       Others    Military - Sudanese Air Force   
4847       Others                           Others   
4892       Others                           Others   
5001  Philippines  Military - Philippine Air Force   

                             route   ac_type all_aboard PASAJEROS A BORDO  \
4                         Sin_Dato    Others         20                 ?   
5                         Sin_Dato    Others         28                 ?   
7                         Sin_Dato    Others         19                 ?   
8                         Sin_Dato    Others         20                 ?   
9                         Sin_Dato    Others         28                 ?   
...                            ...       ...        ...               ...   
4830                      Sin_Dato    Others         23                 ?   
4833                      Sin_Dato   Antonov         22                16   
4847                      Sin_Dato    Others         12                 ?   
4892                      Training    Others         19                21   
5001  Cagayan de Oro-Lumbia - Jolo  Lockheed         96                88   

     crew_aboard cantidad de fallecidos passenger_fatalities crew_fatalities  \
4              ?                     14                    ?               ?   
5              ?                     28                    ?               ?   
7              ?                     19                    ?               ?   
8              ?                     20                    ?               ?   
9              ?                     27                    ?               ?   
...          ...                    ...                  ...             ...   
4830           ?                     18                    ?               ?   
4833           6                     15                    ?               ?   
4847           ?                     12                    ?               ?   
4892           2                     18                    ?               ?   
5001           8                     50                    ?               ?   

     ground  year  decade  
4         0  1913       1  
5         0  1913       1  
7         0  1915       1  
8         0  1916       1  
9         0  1916       1  
...     ...   ...     ...  
4830      0  2011      11  
4833      0  2012      11  
4847      4  2012      11  
4892      0  2014      11  
5001      3  2021      12  

[221 rows x 13 columns]

In [293]:
registros_con_interrogante = aa[aa['crew_aboard'] == '?']
registros_con_interrogante

Ruta                OperadOR                 route    ac_type  \
4      Others  Military - German Navy              Sin_Dato     Others   
5     Germany  Military - German Navy              Sin_Dato     Others   
7     Germany  Military - German Navy              Sin_Dato     Others   
8     England  Military - German Navy              Sin_Dato     Others   
9     Germany  Military - German Army              Sin_Dato     Others   
...       ...                     ...                   ...        ...   
4769   Russia                  Others              Sin_Dato  Mil Mi-8T   
4812   Others    Military - U.S. Army              Sin_Dato     Boeing   
4830    India                  Others              Sin_Dato     Others   
4847   Others                  Others              Sin_Dato     Others   
4940   Others                  Others  Mogadishu - Djibouti     Others   

     all_aboard PASAJEROS A BORDO crew_aboard cantidad de fallecidos  \
4            20                 ?           ?                     14   
5            28                 ?           ?                     28   
7            19                 ?           ?                     19   
8            20                 ?           ?                     20   
9            28                 ?           ?                     27   
...         ...               ...         ...                    ...   
4769         10                 ?           ?                      7   
4812         38                 ?           ?                     38   
4830         23                 ?           ?                     18   
4847         12                 ?           ?                     12   
4940         72                72           ?                      1   

     passenger_fatalities crew_fatalities ground  year  decade  
4                       ?               ?      0  1913       1  
5                       ?               ?      0  1913       1  
7                       ?               ?      0  1915       1  
8                       ?               ?      0  1916       1  
9                       ?               ?      0  1916       1  
...                   ...             ...    ...   ...     ...  
4769                    ?               ?      0  2010      11  
4812                    ?               ?      0  2011      11  
4830                    ?               ?      0  2011      11  
4847                    ?               ?      4  2012      11  
4940                    1               ?      0  2016      11  

[202 rows x 13 columns]

In [294]:
registros_con_interrogante = aa[aa['PASAJEROS A BORDO'] == '?']
registros_con_interrogante

Ruta                OperadOR     route    ac_type all_aboard  \
4      Others  Military - German Navy  Sin_Dato     Others         20   
5     Germany  Military - German Navy  Sin_Dato     Others         28   
7     Germany  Military - German Navy  Sin_Dato     Others         19   
8     England  Military - German Navy  Sin_Dato     Others         20   
9     Germany  Military - German Army  Sin_Dato     Others         28   
...       ...                     ...       ...        ...        ...   
4765   Others                  Others  Sin_Dato       Mi-8         28   
4769   Russia                  Others  Sin_Dato  Mil Mi-8T         10   
4812   Others    Military - U.S. Army  Sin_Dato     Boeing         38   
4830    India                  Others  Sin_Dato     Others         23   
4847   Others                  Others  Sin_Dato     Others         12   

     PASAJEROS A BORDO crew_aboard cantidad de fallecidos  \
4                    ?           ?                     14   
5                    ?           ?                     28   
7                    ?           ?                     19   
8                    ?           ?                     20   
9                    ?           ?                     27   
...                ...         ...                    ...   
4765                 ?           ?                     28   
4769                 ?           ?                      7   
4812                 ?           ?                     38   
4830                 ?           ?                     18   
4847                 ?           ?                     12   

     passenger_fatalities crew_fatalities ground  year  decade  
4                       ?               ?      0  1913       1  
5                       ?               ?      0  1913       1  
7                       ?               ?      0  1915       1  
8                       ?               ?      0  1916       1  
9                       ?               ?      0  1916       1  
...                   ...             ...    ...   ...     ...  
4765                    ?               ?      0  2010      11  
4769                    ?               ?      0  2010      11  
4812                    ?               ?      0  2011      11  
4830                    ?               ?      0  2011      11  
4847                    ?               ?      4  2012      11  

[204 rows x 13 columns]

EN LAS COLUMNAS DE PASAJEROS Y TRIPULANTES TANTO A BORDO COMO FALLECIDOS HAY MAS DE 200 REGISTROS CON INTERROGANTE PARA NO PERDER LA COLUMNA CANTIDAD DE FALLECIDOS SE MANTIENEN ESOS INDICES PERO CON EL VALOR SIN DATO

In [295]:
def conservar_sindato(texto):
    if '?' in texto:
        return 'Sin_Dato'
    else:
        return texto

aa['PASAJEROS A BORDO'] = aa['PASAJEROS A BORDO'].apply(conservar_sindato)

In [296]:
def conservar_sindato(texto):
    if '?' in texto:
        return 'Sin_Dato'
    else:
        return texto

aa['crew_aboard'] = aa['crew_aboard'].apply(conservar_sindato)

In [297]:
def conservar_sindato(texto):
    if '?' in texto:
        return 'Sin_Dato'
    else:
        return texto

aa['passenger_fatalities'] = aa['passenger_fatalities'].apply(conservar_sindato)

In [299]:
aa['crew_fatalities'] = aa['crew_fatalities'].astype(str)

In [300]:
def conservar_sindato(texto):
    if '?' in texto:
        return 'Sin_Dato'
    else:
        return texto

aa['crew_fatalities'] = aa['crew_fatalities'].apply(conservar_sindato)

In [310]:
registros_con_interrogante = aa[aa['ground'] == '?']
registros_con_interrogante

Ruta                                    OperadOR  \
219        Others                                      Others   
220          EEUU                       Pacific Air Transport   
313         China         China National Aviation Corporation   
315          EEUU                            United Air Lines   
461        Others                                      Others   
484        Others                                      Others   
517        Others                                     Panagra   
555       England                             British Airways   
652         China         China National Aviation Corporation   
752        Others                        Military - U.S. Army   
794   Philippines                 Australian National Airways   
861        Russia                                    Aeroflot   
1040       Russia                                    Aeroflot   
1218       Others                                  Air France   
1332      England                                      Others   
1347         EEUU  Military - U.S. Air Force / U.S. Air Force   
1387       Others                                      Others   
1395       Russia                                    Aeroflot   
1422         EEUU                                      Others   
1538       Others                                      Others   
1542       Others                                     Private   
1639       Canada                                      Others   
1747       Others                                      Others   
1885       Brazil                                      Others   
2028       Others                                    Sin_Dato   
2129         EEUU                                      Others   
2387       Others                                TAME Ecuador   
2954       Others                          Military - US Navy   
4753       Others                                 Service Air   
4783       Russia                          Kamchatka Airlines   
4931         EEUU                                      Others   
4999       Others                                      Others   

                                                  route  \
219                                            Sin_Dato   
220                           Burbank, CA - Oakland, CA   
313                                   Canton - Shanghai   
315                            Seattle, WA - Dallas, TX   
461                                     Krakow - Warsaw   
484                  San José  San Isidro de El General   
517                                            Sin_Dato   
555                                            Training   
652                                            Sin_Dato   
752                               Port Moreaby - Nadzab   
794                                            Sin_Dato   
861                               Riga, Latvia - Moscow   
1040                             Zyryanka - Srednekymsk   
1218                 Saigon - Karachi - Bahrain - Paris   
1332                                  Croydon - Le Mans   
1347                                           Training   
1387                                      Cairo - Luxor   
1395  Magadan - Okhotsk - Nikolayevsk-on-Amur - Khab...   
1422                            Teterboro - Saint Louis   
1538                                           Sin_Dato   
1542                              Tampico - Mexico City   
1639                       Great Whale River – Val-d’Or   
1747                               Kuwait City - Tehran   
1885                              Sao Paulo - Concordia   
2028                                           Sin_Dato   
2129                                           Training   
2387                                           Sin_Dato   
2954                                 Military exercises   
4753                                       Goma - Isiro   
4783                                    Elizovo - Dukup   
4931                       Sightseeing

In [311]:
def conservar_sindato(texto):
    if '?' in texto:
        return 'Sin_Dato'
    else:
        return texto

aa['ground'] = aa['ground'].apply(conservar_sindato)

In [312]:
registros_con_interrogante = aa[aa['ground'] == '?']
registros_con_interrogante

Empty DataFrame
Columns: [Ruta, OperadOR, route, ac_type, all_aboard, PASAJEROS A BORDO, crew_aboard, cantidad de fallecidos, passenger_fatalities, crew_fatalities, ground, year, decade]
Index: []

In [314]:
conteoground = aa['ground'].value_counts()
conteoground

ground
0           4714
1             63
2             34
Sin_Dato      32
3             21
4             16
5             12
7             10
8              9
10             6
6              6
14             5
13             4
22             4
19             4
11             4
24             3
44             3
20             3
12             3
2750           2
25             2
15             2
30             2
125            2
35             2
9              1
49             1
85             1
40             1
39             1
18             1
225            1
78             1
50             1
45             1
31             1
32             1
113            1
71             1
33             1
23             1
87             1
17             1
29             1
16             1
36             1
58             1
63             1
37             1
53             1
47             1
Name: count, dtype: int64

In [320]:
dante = aa[aa['ground'] == '2750']
dante

Ruta           OperadOR                 route ac_type all_aboard  \
4317  EEUU  American Airlines  Boston - Los Angeles  Boeing         92   
4322  EEUU   United Air Lines  Boston - Los Angeles  Boeing         65   

     PASAJEROS A BORDO crew_aboard cantidad de fallecidos  \
4317                81          11                     92   
4322                56           9                     65   

     passenger_fatalities crew_fatalities ground  year  decade  
4317                   81              11   2750  2001      10  
4322                   56               9   2750  2001      10

EN ESTOS INDICES DE ARRIBA SE ENCUENTRA UN DATO ATIPICO PARA LA COLUMNA GROUND Y SE PROCEDE A CAMBIAR POR SIN DATO

In [321]:
aa.at[4317, 'ground'] = 'Sin_Dato'
aa.at[4322, 'ground'] = 'Sin_Dato'

In [323]:
conteoground = aa['ground'].value_counts()
conteoground

ground
0           4714
1             63
Sin_Dato      34
2             34
3             21
4             16
5             12
7             10
8              9
10             6
6              6
14             5
13             4
22             4
19             4
11             4
24             3
44             3
20             3
12             3
25             2
15             2
30             2
125            2
35             2
49             1
40             1
9              1
39             1
85             1
32             1
45             1
50             1
78             1
18             1
225            1
17             1
113            1
71             1
33             1
23             1
87             1
31             1
29             1
16             1
36             1
58             1
63             1
37             1
53             1
47             1
Name: count, dtype: int64

Debes graficar y medir el KPI propuesto a continuación, representándolo adecuadamente en el dashboard. A su vez, tambíen tienes que proponer, medir y graficar un segundo KPI que consideres relevante para la temática. El KPI propuesto es:

Evaluar la disminución de un 10% la tasa de fatalidad de la tripulación en los últimos 10 años, comparado a la década anterior.

Definimos la tasa de fatalidad de la tripulación como el número total de tripulantes fallecidos en los accidentes registrados en la década a considerar, dividido en la cantidad total de accidentes aéreos ocurridos en este período de tiempo. Su fórmula es (Suma total de fallecidos en el período de tiempo / Suma total de accidentes en el período de tiempo).

In [325]:
dante = aa[aa['ground'] == '125']
dante

Ruta           OperadOR                          route ac_type  \
2230  Others  Flying Tiger Line                       Sin_Dato  Others   
4316    EEUU  American Airlines  Washington D.C. - Los Angeles  Boeing   

     all_aboard PASAJEROS A BORDO crew_aboard cantidad de fallecidos  \
2230          4                 0           4                      4   
4316         64                58           6                     64   

     passenger_fatalities crew_fatalities ground  year  decade  
2230                    0               4    125  1966       6  
4316                   58               6    125  2001      10

In [326]:
aa.to_csv('aa.csv', index = False)